lapce-ui/src/problem.rs

```diff
@@ -84,98 +83,157 @@ impl ProblemContent {
-                            EditorLocation {
-                                path: path.clone(),
-                                position: Some(d.diagnostic.range.start),
-                                scroll_offset: None,
-                                history: None,
-                            },
-                        ),
-                        Target::Widget(data.id),
-                    ));
-                    return;
-                }
-                i += msg_lines;
+        // Skip to clicked diagnostic
+        let mut it = diagnostics.into_iter().peekable();
+        while let Some(file_diagnostic) = it.peek() {
+            // Is current diagnostic the clicked one?
+            if line_cursor + file_diagnostic.lines < click_line {
+                // No. Move line cursor and consume diagnostic
+                line_cursor += file_diagnostic.lines;
+                it.next();
+            } else {
+                // We found diagnostic we are looking for
+                break;
+            }
+        }
 
-                for related in d
-                    .diagnostic
-                    .related_information
-                    .as_ref()
-                    .unwrap_or(&Vec::new())
-                {
-                    let lines = related.message.matches('\n').count() + 1 + 1;
-                    if i <= n && n <= i + lines {
-                        ctx.submit_command(Command::new(
-                            LAPCE_UI_COMMAND,
-                            LapceUICommand::JumpToLocation(
-                                None,
-                                EditorLocation {
-                                    path: related
-                                        .location
-                                        .uri
-                                        .to_file_path()
-                                        .unwrap(),
-                                    position: Some(related.location.range.start),
-                                    scroll_offset: None,
-                                    history: None,
-                                },
-                            ),
-                            Target::Widget(data.id),
-                        ));
-                        return;
-                    }
-                    i += lines;
-                }
+        // Handle current diagnostic
+        let file_diagnostic = it.next().expect("Editor diagnostic not found. We should find here file diagnostic but nothing left in the array. Please report a bug");
+
+        if line_cursor > click_line {
+            log::error!(
+                "Line cursor is larger than clicked line. This should never happen!"
+            );
+            return;
+        }
+
+        let msg_lines = file_diagnostic.diagnostic.message.lines().count();
+
+        // Widget has mouse about it and line is clicked one.
+        if ctx.is_hot()
+            && (line_cursor..(line_cursor + msg_lines)).contains(&click_line)
```

Comment:
Should it be ```(line_cursor..=(line_cursor + msg_lines)).contains(&click_line)``` here?

Reply:
No, this is this section:

![Screenshot_20220618-115721_Firefox.jpg](https://user-images.githubusercontent.com/970831/174430579-8a1707ff-a62a-4aa5-b33c-449f7461b014.jpg)

So this is either current line or not. Knowing this is exclusively range it could be simply `line_cursor == click_line` if it will always be one line.

Reply:
If you load https://github.com/lapce/lapce-rust, the last line cause a crash